libvast/vast/type.hpp

```diff
@@ -1135,8 +1094,13 @@ struct formatter<vast::type> {
           out = format_to(out, "{}", x);
         },
         value);
-    if (const auto& attributes = value.attributes(); !attributes.empty())
-      out = format_to(out, " {}", fmt::join(attributes, " "));
+    for (bool first = false; const auto& attribute : value.attributes()) {
+      if (!first) {
+        out = format_to(out, " ");
+        first = false;
+      }
+      out = format_to(out, "{}", attribute);
+    }
```

Comment:
{fmt} 7 and 8 use `*it++` in `fmt::join(it, end, delim)`, which is not working will all valid C++20 ranges, where the postfix increment operator is allowed to return `void`. It's fixed on {fmt} master, so for now it'll have to be this ugly.